wordpress-mobile/WordPress-Android
# Add initial Quick Start card item implementation to the improved My Site fragment

Fixes #13583

This PR introduces a basic implementation of the Quick Start card item. This basic implementation can be divided into 3 parts:

1. The Quick Start card's title bar, containing the progress bar on the left side, the overflow button on the right side, and the title in the middle.
1. The inner list of task cards that can be scrolled horizontally.
1. The bottom sheet menu that appears when the overflow button is clicked.

Items 1 and 2 are handled by this PR, while item 3 is handled by #13725. The UI of the nested task card and the necessary logic to show each task will be handled in future PRs. For testing purposes, this PR includes two static "sample" cards based on the design spec (more details on the linked issue):

https://user-images.githubusercontent.com/830056/104061781-01357b00-51d8-11eb-999f-bb264563dc80.mp4

Note:

In order to achieve the exact same design of the progress component that is depicted in the design spec, I initially considered [updating our Material Components library](https://github.com/wordpress-mobile/WordPress-Android/pull/13634) so we could use the new `ProgressIndicator` component. Later on, I decided to use a regular `ProgressBar` with a few tweaks, instead. I gave more details on this decision [here](https://github.com/wordpress-mobile/WordPress-Android/pull/13634#issuecomment-756477352). The most noticeable difference that comes with this is that the progress bar does not contain rounded corners. Hopefully we'll be able to replace it with a `ProgressIndicator` once it hits a stable version of the Material Components library.

### To test

Make sure all unit tests in `MySiteViewModelTest` pass. 

Additionally:

1. Turn on the Improved My Site flag and restart the app.
1. Go to the My Site screen.
1. Notice the 2 sample Quick Start cards.
1. Make sure nothing looks out of place.
1. Close the app and open it again.
1. Notice how the progress bar animates from 0 to its actual value (this happens pretty fast, so make sure not to miss it).

---

PR submission checklist:
